kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller_test.go

```diff
@@ -95,6 +93,131 @@ var _ = Describe("TaskRunToolCall Controller", func() {
 			utils.ExpectRecorder(recorder).ToEmitEventContaining("ExecutionSucceeded")
 		})
 	})
+
+	Context("Pending -> Failed", func() {
+		It("fails when arguments are invalid", func() {
+			teardown := setupTestAddTool(ctx)
+			defer teardown()
+
+			// Create TaskRunToolCall with Pending status but invalid arguments
+			taskRunToolCall := &TestTaskRunToolCall{
+				name:      "invalid-args-trtc",
+				toolName:  addTool.name,
+				arguments: `invalid json`, // Invalid JSON
+			}
+			defer taskRunToolCall.Teardown(ctx)
+
+			trtc := taskRunToolCall.SetupWithStatus(ctx, kubechainv1alpha1.TaskRunToolCallStatus{
+				Phase:        kubechainv1alpha1.TaskRunToolCallPhasePending,
+				Status:       "Pending",
+				StatusDetail: "Ready for execution",
+				StartTime:    &metav1.Time{Time: time.Now().Add(-1 * time.Minute)},
+			})
+
+			By("reconciling the taskruntoolcall with invalid arguments")
+			reconciler, recorder := reconciler()
+
+			_, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      trtc.Name,
+					Namespace: trtc.Namespace,
+				},
+			})
+
+			// We expect an error during reconciliation
+			Expect(err).To(HaveOccurred())
+
+			By("checking the taskruntoolcall status is set to Error")
+			updatedTRTC := &kubechainv1alpha1.TaskRunToolCall{}
+			err = k8sClient.Get(ctx, types.NamespacedName{
+				Name:      trtc.Name,
+				Namespace: trtc.Namespace,
+			}, updatedTRTC)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedTRTC.Status.Status).To(Equal("Error"))
+			Expect(updatedTRTC.Status.StatusDetail).To(Equal("Invalid arguments JSON"))
+			Expect(updatedTRTC.Status.Error).NotTo(BeEmpty())
+
+			By("checking that error events were emitted")
+			utils.ExpectRecorder(recorder).ToEmitEventContaining("ExecutionFailed")
+		})
+
+		// Tests for MCP tools without approval requirement
+		Context("Pending -> Succeeded (MCP Tool)", func() {
```

Comment:
is there a reason this is indented under the other context?

Reply:
Nope, bad copypasta, fixin'er